spec/integration/show_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'capybara/rails'
+require 'capybara/minitest'
+
+RSpec.describe 'User show page', type: :system do
+  let!(:user) do
+    User.create(name: 'John', photo: 'https://example.com/profile.jpg', bio: 'Software Engineer')
+    User.create(name: 'Daniel', photo: 'https://example.com/profile.jpg', bio: 'Electrical Engineer')
+  end
+
+  let!(:posts) do
+    user.posts.create([
+                        { text: 'First post', comments_counter: 2, likes_counter: 5 },
+                        { text: 'Second post', comments_counter: 3, likes_counter: 7 }
+                      ])
+  end
+
+  it 'displays user information and posts' do
+    visit user_path(user)
+    expect(page).to have_content('John')
+    expect(page).to have_content("Number of post: #{user.posts_counter}")
+
+    expect(page).to have_content('Bio :')
+    expect(page).to have_content('Software Engineer')
+  end
```

Comment:
- [ ] This test scenario fails whenever I run it. It expects to see the text _**John**_ in the `User#Show` page for **_Daniel's_** profile. When you created the users in the `let` block, the last profile that will be returned is that of **_Daniel_**. Therefore, the variable user will be _**Daniel's**_ profile. To fix this, I would advise you to use `instance variables (@user1, @user2)` to store the user profiles and use them where needed. 👍 💻 

![image](https://github.com/Kaghenimbale/blog-app/assets/77106962/991fab1b-62c4-484b-add9-ae8185148acf)